pkg/controllers/managementuser/resourcequota/resource_quota_sync.go

```diff
@@ -438,7 +438,10 @@ func completeQuota(requestedQuota *v32.ResourceQuotaLimit, defaultQuota *v32.Res
 		return nil, err
 	}
 	for key, value := range requestedQuotaMap {
-		newLimitMap[key] = value
+		// Only override the values for keys (resources) that actually exist in the project quota.
+		if newLimitMap[key] != nil {
```

Comment:
This won't work because the string keys are always present in the map (created by converting the ResourceQuotaLimit value), but the values may or may not be nil. This would change the logic.